apps/cron/src/tasks/processGemsPayout/createTopReferrerRewardsContract.ts

```diff
@@ -0,0 +1,62 @@
+import { prisma } from '@charmverse/core/prisma-client';
+import { getCurrentSeasonStart, getLastWeek } from '@packages/dates/utils';
+import { getTopConnectorOfTheDay } from '@packages/scoutgame/topConnector/getTopConnectors';
+import { DateTime } from 'luxon';
+import { parseUnits } from 'viem';
+import { optimism } from 'viem/chains';
+
+import { optimismTokenDecimals, optimismTokenAddress } from './createNewScoutRewardsContract';
+import { createSablierAirdropContract } from './createSablierAirdropContract';
+
+const TOP_REFERRER_REWARDS_AMOUNT = parseUnits('25', optimismTokenDecimals);
+
+export async function createTopReferrerRewardsContract() {
+  const topConnectors: { address: string; date: DateTime }[] = [];
+  const week = getLastWeek();
+  const season = getCurrentSeasonStart(week);
+
+  for (let day = 1; day <= 7; day++) {
+    const date = DateTime.utc().minus({ days: day });
+    const topConnector = await getTopConnectorOfTheDay({ date });
+
+    if (topConnector) {
+      topConnectors.push({ address: topConnector.address, date });
+    }
+  }
+
+  const { hash, contractAddress, cid } = await createSablierAirdropContract({
+    adminPrivateKey: process.env.OP_AIRDROP_ADMIN_PRIVATE_KEY as `0x${string}`,
+    campaignName: `Top Referrer Rewards Season: ${season}, Week: ${week}`,
+    chainId: optimism.id,
+    tokenAddress: optimismTokenAddress,
+    tokenDecimals: optimismTokenDecimals,
+    recipients: topConnectors.map(({ address }) => ({ address: address as `0x${string}`, amount: 25 }))
+  });
+
+  await prisma.partnerRewardPayoutContract.create({
+    data: {
+      chainId: optimism.id,
+      contractAddress,
+      season,
+      week,
+      cid,
+      tokenAddress: optimismTokenAddress,
+      tokenDecimals: optimismTokenDecimals,
+      partner: 'optimism_top_referrer',
```

Comment:
at first i left a comment saying we shouldnt make 'optimism_top_referrer' as a partner, but we do use 'partner reward' to describe it in the figma, so i think it makes sense